src/native/libs/System.Security.Cryptography.Native/pal_evp.c

```diff
@@ -291,100 +315,18 @@ int32_t CryptoNative_EvpMdSize(const EVP_MD* md)
     return EVP_MD_get_size(md);
 }
 
-const EVP_MD* CryptoNative_EvpMd5(void)
-{
-    pthread_once(&g_evpFetch, EnsureFetchEvpMdAlgorithms);
-    return g_evpFetchMd5;
-}
-
-const EVP_MD* CryptoNative_EvpSha1(void)
-{
-    // No error queue impact.
-    return EVP_sha1();
-}
-
-const EVP_MD* CryptoNative_EvpSha256(void)
-{
-    // No error queue impact.
-    return EVP_sha256();
-}
-
-const EVP_MD* CryptoNative_EvpSha384(void)
-{
-    // No error queue impact.
-    return EVP_sha384();
-}
-
-const EVP_MD* CryptoNative_EvpSha512(void)
-{
-    // No error queue impact.
-    return EVP_sha512();
-}
-
-const EVP_MD* CryptoNative_EvpSha3_256(void)
-{
-    // No error queue impact.
-#if HAVE_OPENSSL_SHA3
-    if (API_EXISTS(EVP_sha3_256))
-    {
-        return EVP_sha3_256();
-    }
-#endif
-
-    return NULL;
-}
-
-const EVP_MD* CryptoNative_EvpSha3_384(void)
-{
-    // No error queue impact.
-#if HAVE_OPENSSL_SHA3
-    if (API_EXISTS(EVP_sha3_384))
-    {
-        return EVP_sha3_384();
-    }
-#endif
-
-    return NULL;
-}
-
-const EVP_MD* CryptoNative_EvpSha3_512(void)
-{
-    // No error queue impact.
-#if HAVE_OPENSSL_SHA3
-    if (API_EXISTS(EVP_sha3_512))
-    {
-        return EVP_sha3_512();
-    }
-#endif
-
-    return NULL;
-}
-
-const EVP_MD* CryptoNative_EvpShake128(void)
-{
-    // No error queue impact.
-#if HAVE_OPENSSL_SHA3
-    if (API_EXISTS(EVP_shake128))
-    {
-        return EVP_shake128();
-    }
-#endif
-
-    return NULL;
-}
-
-const EVP_MD* CryptoNative_EvpShake256(void)
-{
-    // No error queue impact.
-#if HAVE_OPENSSL_SHA3
-    if (API_EXISTS(EVP_shake256))
-    {
-        return EVP_shake256();
-    }
-#endif
-
-    return NULL;
-}
+// MD5 should use a non-FIPS implementation if it is available. We should not fail
+// to fetch MD5 even on a FIPS enforced system.
+SETUP_MD_FETCH(EvpMd5, EVP_md5, "MD5", "-fips")
```

Comment:
Text search for `CryptoNative_EvpMd5` won't show this location anymore with this change. I think it would be better to pass the whole name to the macro.